prime-router/docs/proposals/proposal-process-readme.md

```diff
@@ -0,0 +1,23 @@
+## Background
+
+This document is a description of the engineering proposal process as a proposal.
+
+## Goals
+
+The proposal process aims to promote detailed discussion on significant engineering changes. Putting these designs in writing allows for a better review. Reviewers can comment on the proposal as a whole as well as specific small details. With this up-front review, we can improve a design early and avoid spending coding time on the wrong plan.
+
+At the same time, there is a balance between the spent on paper design and writing code. There is a limit to the types of problems a paper design can flush out. We do not expect a design proposal to have all aspects of a design detailed. Every engineer should judge how much detail to put into their documents. After their pull-request is approved, the engineer should feel free to improve a design as they code.
+
+## Proposal
+
+The proposal process involves a Markdown document for the proposal, and a GitHub pull-request that will contain the discussion about the proposal. Using the project's public GitHub infrastructure allows for outside reviewers.
+
+**Steps**
+
+1. Put a proposal document in a single markdown document in the project's proposal directory. Please include sections for context and motivation behind the proposal, the goals of the proposal, and the details of the proposal itself. Put assets in a subdirectory or the assets directory.
```

Comment:
Having a template for this would be extremely helpful. [Rust’s RFC process](https://github.com/rust-lang/rfcs) has been the general standard I’ve seen a lot of large open-source projects base their projects on (it’s been a pretty successful model). They’ve got a [good template](https://github.com/rust-lang/rfcs/blob/master/0000-template.md), although I think their process is definitely too heavy-weight for the scale (and speed) we need to operate at here. Their template can probably offer more utility than the rest for our case here.

> Put assets in a subdirectory or the assets directory

I think you should recommend one specific structure here, otherwise people will do different things, and it will get messy. I suggest adding a directory at `docs/proposals/<proposal_name>-assets/` for assets.

Reply:
Or as @jimduff-usds is doing in #320, proposals with assets should be a directory:

```
docs/
└─┬ proposals/
  ├── <proposal_with_no_assets_name>.md
  └─┬ <proposal_with_assets_name>/
    ├── README.md (proposal text)
    ├── <asset1>
    ├── <asset2>
    └── etc.
```

Reply:
I agree that the Rust RFC process is probably too heavyweight at this point, but we could crib a few things from it. I do especially appreciate the value of an RFC being open for ten days so everyone has a chance to look and provide feedback before it's adjudicated.

Reply:
We are not a project that is ready for a formal RFC process. 

I like Jim's sub-folders.

Reply:
As another quick tidbit to steal from Rust, prefixing the proposal with a number or date (a sort-friendly one, like “2021-01-19”) would be helpful here, too. For example, we currently have a proposal named `authentication.md`, but if we made major, proposal-worthy authentication changes in the future, the prefix would be helpful in making it clear which proposal came first.